src/main/java/com/faforever/client/units/UnitsController.java

```diff
@@ -41,7 +46,14 @@ public void onDisplay(NavigateEvent navigateEvent) {
 
   private void loadUnitDataBase(UnitDataBaseType newValue) {
     UnitDatabase unitDatabase = clientProperties.getUnitDatabase();
-    unitsRoot.getEngine().load(newValue == UnitDataBaseType.SPOOKY ? unitDatabase.getSpookiesUrl() : unitDatabase.getRackOversUrl());
+
+    String rackoverSettingsEncoded = "";
+    if (newValue == UnitDataBaseType.RACKOVER) {
+      String rackoverSettings = new Gson().toJson(new RackoversDBSettings(i18n.getUserSpecificLocale()));
+      rackoverSettingsEncoded = Base64.getEncoder().encodeToString(rackoverSettings.getBytes());
+    }
+
+    unitsRoot.getEngine().load(newValue == UnitDataBaseType.SPOOKY ? unitDatabase.getSpookiesUrl() : String.format(unitDatabase.getRackOversUrl(), rackoverSettingsEncoded));
```

Comment:
In the `if` we check if it's Rackover's, here we check if it's SPOOKY's and assume it's Rackover's otherwise.
One of the checks is redundant, and it only works as long as there are only two values.

I suggest we refactor this to use a switch-case with a default case that throws a `ProgrammingException` if `newValue` is an unimplemented enum value.